Helper/Order.php

```diff
@@ -731,6 +718,208 @@ public function saveUpdateOrder($reference, $boltTraceId = null, $hookType = nul
+            $order->addStatusHistoryComment(
+                __('Bolt transaction: %1', $this->formatReferenceUrl($transaction->reference))
+            );
+        }
+
+        return $order;
+    }
+
+    /**
+     * @param $order
+     * @param $transaction
+     * @return bool
+     * @throws \Exception
+     */
+    public function updateExistingOrder($order, $transaction)
+    {
+        /** @var OrderModel $order */
+        $subtotal = (int) ($order->getSubtotal() * 100);
+        $taxAmount = (int) ($order->getTaxAmount() * 100);
+        $shippingAmount = (int) ($order->getShippingAmount() * 100);
+        $totalAmount = (int) ($order->getGrandTotal() * 100);
+
+        $transactionSubtotalAmount = $transaction->order->cart->subtotal_amount->amount;
+        $transactionTaxAmount = $transaction->order->cart->tax_amount->amount;
+        $transactionShippingAmount = $transaction->order->cart->shipping_amount->amount;
+        $transactionGrandTotalAmount = $transaction->order->cart->total_amount->amount;
+
+        // If totals differs then delete order.
+        if (abs($subtotal - $transactionSubtotalAmount) <= self::MISMATCH_TOLERANCE
+            && abs($taxAmount - $transactionTaxAmount) <= self::MISMATCH_TOLERANCE
+            && abs($shippingAmount - $transactionShippingAmount) <= self::MISMATCH_TOLERANCE
+            && abs($totalAmount - $transactionGrandTotalAmount) <= self::MISMATCH_TOLERANCE
+        ) {
+            return false;
+        }
+
+        try {
+            $order->cancel()->save()->delete();
+        } catch (\Exception $e) {
+            $order->delete();
+        }
+
+        return true;
+    }
+
+    /**
+     * @param $displayId
+     * @throws \Exception
+     */
+    public function deleteOrderByIncrementId($displayId)
+    {
+        list($incrementId, $quoteId) = array_pad(
+            explode(' / ', $displayId),
+            2,
+            null
+        );
+
+        $order = $this->getExistingOrder($incrementId);
+        if ($order && $order->getState() === OrderModel::STATE_NEW) {
+            $order->cancel()->save()->delete();
+        }
+    }
+
+    /**
+     * @param $orderIncrementId
+     * @return OrderModel|null
+     */
+    private function getExistingOrder($orderIncrementId)
+    {
+        /** @var OrderModel $order */
+        $order = $this->cartHelper->getOrderByIncrementId($orderIncrementId);
+
+        return ($order && $order->getId()) ? $order : null;
+    }
+
+    /**
+     * @param $immutableQuote
+     * @param $transaction
+     * @return Quote
+     * @throws LocalizedException
+     * @throws NoSuchEntityException
+     * @throws \Magento\Framework\Exception\AlreadyExistsException
+     * @throws \Magento\Framework\Exception\SessionException
+     * @throws \Zend_Validate_Exception
+     */
+    public function prepareQuote($immutableQuote, $transaction)
+    {
+        // Load and prepare parent quote
```

Comment:
Prepare quote is already in the function name. Maybe add a short description of what preparing the quote entails, or remove comment since reading the code is pretty self-explanatory.

Reply:
done